lib/chef/resource.rb

```diff
@@ -1414,8 +1442,31 @@ def self.new_action_provider_class
         use_inline_resources
         include_resource_dsl true
         define_singleton_method(:to_s) { "#{resource_class} action provider" }
-        define_singleton_method(:inspect) { to_s }
-        define_method(:load_current_resource) {}
+        def self.inspect
+          to_s
+        end
+        def load_current_resource
+          if new_resource.respond_to?(:load_current_value!)
+            # dup the resource and then reset desired-state properties.
+            current_resource = new_resource.dup
+
+            # We clear desired state in the copy, because it is supposed to be actual state.
+            # We keep identity properties and non-desired-state, which are assumed to be
+            # "control" values like `recurse: true`
+            current_resource.class.properties.each do |name,property|
+              if property.desired_state? && !property.identity? && !property.name_property?
+                property.reset(current_resource)
+              end
+            end
+
+            if current_resource.method(:load_current_value!).arity > 0
+              current_resource.load_current_value!(new_resource)
```

Comment:
I don't remember seeing this in the doc changes. What are the cases when you'd want to have this explicitly passed?